### ISSUE TITLE
Create API endpoint for pictures

### ISSUE THREAD

**Issue by Vallenain** [vpmalley/jetlag#53]
`api/<version>/pictures` I guess. And so we might rename the `api/<version>/pix/upload` into `api/<version>/pictures/upload`, what do you think ?

**Comment by Vallenain** [comment]
Ok, just realized I said the opposite in my email...
So, do we create a standard endpoint with GET/POST/PUT for Pictures or not ? If Pictures can only be nested objects ( = sub element), I don't think we need an endpoint.
As long as the upload endpoint returns the updated Picture object when successful, it should be good enough.

**Comment by vpmalley** [comment]
Yeah I think in the long term that would be a good endpoint to have, but if the upload returns the picture that should be enough for now

**Comment by vpmalley** [comment]
There is now an endpoint for pictures. So far, there is no author linked to a picture, i.e. every user can create, update and delete a picture.
It is possible to create/update a picture with its linked objects : links and place.

I let you try it and let me know if that's what you expect

**Comment by vpmalley** [comment]
TODO
- [ ] apply authoring

**Comment by Vallenain** [comment]
I have done some refactoring of the code related to paragraph edition and I really think it'd be easier and more logical to have a /api/pix/upload which takes one file and return a full Picture object.
2 reasons:
- we avoid sending two requests to the server for what's seem to be one task
- we avoid creating a Picture object with no properties (since no picture has been uploaded so far) which whould remain as ghost if the user closes its browser at this stage.

**Comment by vpmalley** [comment]
yes I suppose it makes sense.

**Comment by vpmalley** [comment]
I changed the endpoints to do what you expected:
- /pix/upload uploads a new picture (taking the `file` and `caption` arguments) and returns a new picture
- /pix/{id}/upload uploads a file for an existing picture (taking the `file` and `caption` arguments) and returns the existing picture
